README.md

```diff
@@ -1,132 +1,488 @@
-gradle build
+This command will trigger a streaming inspection Dataflow pipeline that will process all the CSV files in the demo-data 
+GCS bucket (specify in _filePattern_ parameter). The inspection findings can be found in the BQ dataset (_dataset_ 
+parameter) table.
+
+#### De-Identification
+
+Run following command to trigger the de-identification pipeline:
 
-gradle run -DmainClass=com.google.swarm.tokenization.DLPTextToBigQueryStreamingV2 -Pargs=" --region=<region> --project=<projct_id> --streaming --enableStreamingEngine --tempLocation=gs://<bucket>/temp --numWorkers=1 --maxNumWorkers=2 --runner=DataflowRunner --filePattern=gs://<path>.csv --dataset=<name>   --inspectTemplateName=<inspect_template> --deidentifyTemplateName=<deid_tmplate> --DLPMethod=DEID"
 ```
-## S3 Scanner
+gradle run -DmainClass=com.google.swarm.tokenization.DLPTextToBigQueryStreamingV2 \
+-Pargs=" --region=${REGION} \
+--project=${PROJECT_ID} \
+--streaming --enableStreamingEngine \
+--tempLocation=gs://${PROJECT_ID}-demo-data/temp \
+--numWorkers=2 --maxNumWorkers=3 \
+--runner=DataflowRunner \
+--filePattern=gs://${PROJECT_ID}-demo-data/*.csv \
+--dataset=${BQ_DATASET_NAME}   \
+--inspectTemplateName=${INSPECT_TEMPLATE_NAME} \
+--deidentifyTemplateName=${DEID_TEMPLATE_NAME} \
+--batchSize=200000\
+--DLPMethod=DEID" 
+```
+
+This command will trigger a streaming de-identification Dataflow pipeline that will process all the CSV files in the 
+demo-data GCS bucket (specify in _filePattern_ parameter). The de-identified results can be found in the BQ dataset 
+(_dataset_ parameter) tables with the same names as input files, respectively.
 
+You can run some quick validations for a BigQuery table to check the tokenized data. Following steps use sample dataset 
+to validate de-identified results:
+
+1. In Cloud Shell, display the header row of the CSV file that you used to create the schema:
+```commandline
+gsutil cp gs://${PROJECT_ID}-demo-data/CCRecords_1564602825.csv - | head -1
 ```
-export AWS_ACCESS_KEY="<access_key>"
-export AWS_SECRET_KEY="<secret_key>"
-export AWS_CRED="{\"@type\":\"AWSStaticCredentialsProvider\",\"awsAccessKeyId\":\"${AWS_ACCESS_KEY}\",\"awsSecretKey\":\"${AWS_SECRET_KEY}\"}"
+
+2. In the Query editor of SQL workspace, run the following query to compare the schema to the header row of the CSV file:
+
+```commandline
+SELECT table_name, column_name
+FROM `<dataset_id>.INFORMATION_SCHEMA.COLUMNS`
+WHERE table_name="CCRecords_1564602825"
 ```
+
+There are no spaces in the column names because the pipeline ensures that the column and table names only contains valid 
+characters according to the BigQuery naming standard.
+
+3. Validate that the number of rows in the input file and the output table are equal:
+
+Input records:
+```commandline
+echo $(($(gcloud storage cat gs://${PROJECT_ID}-demo-data/CCRecords_1564602825.csv | wc -l) - 1))
 ```
-gradle spotlessApply
 
-gradle build
+Output records:
+```commandline
+SELECT COUNT(*) as number_of_rows
```

Comment:
Should we use count(distinctI("id")) instead?
Thoughts?

Reply:
"ID" would also work. Using "*" makes it more generic. Also, I didn't find any performance difference between the two.

Reply:
COUNT(*) will consider NULL values too whereas COUNT(ID) will not. We haven't yet encountered a NULL value in results, but it would be better to use COUNT(*).